MVCapplication/src/main/java/com/softserve/if072/mvcapp/controller/ProductPageController.java

```diff
@@ -213,19 +198,124 @@ public String delProduct(@RequestParam int productId){
     @RequestMapping(value = "/stores", method = RequestMethod.GET)
     public String getStoresByProductId(@RequestParam int productId, ModelMap model) {
 
-        final String uri = productUrl + "/{productId}/stores";
+        int userId = getCurrentUser().getId();
+
+        final String getStoresUri = productUrl + "/{productId}/productStores/{userId}";
+        final String getProductUri = productUrl + "/{productId}";
+        final String getAllStoresUri = storeUrl + "/user/{userId}";
+
+        RestTemplate restTemplate = getRestTemplate();
 
         Map<String, Integer> param = new HashMap<>();
         param.put("productId", productId);
+        Product product = restTemplate.getForObject(getProductUri, Product.class, param);
+
+        param.clear();
+        param.put("productId", productId);
+        param.put("userId", userId);
 
-        RestTemplate restTemplate = new RestTemplate();
-        Store[] result = restTemplate.getForObject(uri, Store[].class, param);
-        List<Store> stores = Arrays.asList(result);
+        Store[] arrStores = restTemplate.getForObject(getStoresUri, Store[].class, param);
```

Comment:
Please try this one to resolve issue with array and list
`ResponseEntity<List<Rate>> rateResponse =
        restTemplate.exchange("https://bitpay.com/api/rates",
                    HttpMethod.GET, null, new ParameterizedTypeReference<List<Rate>>() {
            });
List<Rate> rates = rateResponse.getBody();`